riscv/insns/fcvtmod_w_d.h

```diff
@@ -0,0 +1,62 @@
+require_extension('D');
+require_extension(EXT_ZFA);
+require_fp;
+uint64_t a = FRS1_D.v;
+
+uint32_t sign = signF64UI(a);
+uint32_t exp  = expF64UI(a);
+uint64_t frac = fracF64UI(a);
+
+bool inexact = false;
+bool invalid = false;
+
+if (exp == 0) {
+  inexact = (frac != 0);
+  frac = 0;
+} else if (exp == 0x7ff) {
+  /* inf or NaN */
+  invalid = true;
+  frac = 0;
+} else {
+  int true_exp = exp - 1023;
+  int shift = true_exp - 52;
+
+  /* Restore implicit bit.  */
+  frac |= 1ull << 52;
+
+  /* Shift the fraction into place.  */
+  if (shift >= 0) {
+    /* The number is so large we must shift the fraction left.  */
+    if (shift >= 64) {
+      /* The fraction is shifted out entirely.  */
+      frac = 0;
+    } else {
+      frac <<= shift;
+    }
+  } else if (shift > -64) {
+    /* Normal case -- shift right and notice if bits shift out.  */
+    inexact = (frac << (64 + shift)) != 0;
+    frac >>= -shift;
```

Comment:
This looks like undefined behavior because the shift amount exceeds 63.

Reply:
The guard statements are taking care of this:
```
if (shift >= 0) {
} else if (shift > -64) {
}
```
So shift will be [-63 ... -1] and negating it will be [1 ... 63].

I can add an assertion if that is correct spike-style?

Reply:
Best to make the ranges explicit (i.e., `((shift > -64) && (shift < 0))`) to guard against inadvertent reordering of the if/else branches.  The compilers should be smart enough to eliminate some of these checks then.

Reply:
My mistake, I misinterpreted the code! I am fine leaving it alone, or you can clarify it if you prefer.